rtv/subreddit.py

```diff
@@ -73,6 +74,24 @@ def prompt_subreddit(self):
         if name is not None:
             self.refresh_content(order='ignore', name=name)
 
+    @SubredditController.register(Command('SUBREDDIT_FRONTPAGE'))
+    def show_frontpage(self):
```

Comment:
Hmmm, I'm still getting the same error, so maybe my problem wasn't actually the refresh token.
I've copied the token as you suggested.
>>echo $PYTHONPATH
gives:
>> . py.test --record-mode=once --refresh-token=~/refresh_token




Here's the full stack trace when I try to run the tests.
==================================== test session starts ====================================
platform linux -- Python 3.5.1, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: /home/mardi/repos/rtv, inifile: 
collected 111 items 

tests/test_config.py ......
tests/test_content.py .................
tests/test_oauth.py .........
tests/test_objects.py ............................x.
tests/test_page.py ...
tests/test_submission.py ..........
tests/test_subreddit.py ........F
tests/test_subscription.py ......
tests/test_terminal.py .....................

========================================= FAILURES ==========================================
______________________________ test_subreddit_frontpage_toggle ______________________________

subreddit_page = <rtv.subreddit.SubredditPage object at 0x7f54e0bd8860>
terminal = <rtv.terminal.Terminal object at 0x7f54e1056d68>

    def test_subreddit_frontpage_toggle(subreddit_page, terminal):
        with mock.patch.object(terminal, 'prompt_input'):
            terminal.prompt_input.return_value = 'aww'
>           subreddit_page.controller.trigger('/')

tests/test_subreddit.py:184: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
rtv/objects.py:581: in trigger
    return func(self.instance, *args, **kwargs) if func else None
rtv/subreddit.py:75: in prompt_subreddit
    self.refresh_content(order='ignore', name=name)
rtv/subreddit.py:49: in refresh_content
    self.reddit, name, self.term.loader, order=order)
rtv/content.py:431: in from_name
    return cls(display_name, submissions, loader, order=order)
rtv/content.py:365: in __init__
    self.get(0)
rtv/content.py:445: in get
    submission = next(self._submissions)
/usr/lib/python3.5/site-packages/praw/__init__.py:567: in get_content
    page_data = self.request_json(url, params=params)
<decorator-gen-8>:2: in request_json
    ???
/usr/lib/python3.5/site-packages/praw/decorators.py:116: in raise_api_exceptions
    return_value = function(*args, **kwargs)
/usr/lib/python3.5/site-packages/praw/__init__.py:622: in request_json
    retry_on_error=retry_on_error)
/usr/lib/python3.5/site-packages/praw/__init__.py:453: in _request
    response = handle_redirect()
/usr/lib/python3.5/site-packages/praw/__init__.py:434: in handle_redirect
    verify=self.http.validate_certs, **kwargs)
/usr/lib/python3.5/site-packages/praw/handlers.py:146: in wrapped
    result = function(cls, **kwargs)
/usr/lib/python3.5/site-packages/praw/handlers.py:56: in wrapped
    return function(cls, **kwargs)
/usr/lib/python3.5/site-packages/praw/handlers.py:102: in request
    allow_redirects=False, verify=verify)
/usr/lib/python3.5/site-packages/requests/sessions.py:585: in send
    r = adapter.send(request, **kwargs)
/usr/lib/python3.5/site-packages/requests/adapters.py:403: in send
    timeout=timeout
/usr/lib/python3.5/site-packages/requests/packages/urllib3/connectionpool.py:578: in urlopen
    chunked=chunked)
/usr/lib/python3.5/site-packages/requests/packages/urllib3/connectionpool.py:385: in _make_request
    httplib_response = conn.getresponse(buffering=True)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <vcr.patch.VCRRequestsHTTPSConnection/home/mardi/repos/rtv/tests/cassettes/test_subreddit_frontpage_toggle.yaml object at 0x7f54e0dd94e0>
_ = False, kwargs = {'buffering': True}

    def getresponse(self, _=False, **kwargs):
        '''Retrieve the response'''
        # Check to see if the cassette has a response for this request. If so,
        # then return it
        if self.cassette.can_play_response_for(self._vcr_request):
            log.info(
                "Playing response for {0} from cassette".format(
                    self._vcr_request
                )
            )
            response = self.cassette.play_response(self._vcr_request)
            return VCRHTTPResponse(response)
        else:
            if self.cassette.write_protected and self.cassette.filter_request(
                self._vcr_request
            ):
                raise CannotOverwriteExistingCassetteException(
                    "No match for the request (%r) was found. "
                    "Can't overwrite existing cassette (%r) in "
                    "your current record mode (%r)."
                    % (self._vcr_request, self.cassette._path,
>                      self.cassette.record_mode)
                )
E               vcr.errors.CannotOverwriteExistingCassetteException: No match for the request (<Request (GET) https://api.reddit.com/r/aww/.json?limit=1024>) was found. Can't overwrite existing cassette ('/home/mardi/repos/rtv/tests/cassettes/test_subreddit_frontpage_toggle.yaml') in your current record mode ('none').

/usr/lib/python3.5/site-packages/vcr/stubs/__init__.py:232: CannotOverwriteExistingCassetteException
----------------------------------- Captured stderr setup -----------------------------------
2016-07-16 00:48:02,567:DEBUG:cassette.py:54:Entering context for cassette at /home/mardi/repos/rtv/tests/cassettes/test_subreddit_frontpage_toggle.yaml.
----------------------------------- Captured stderr call ------------------------------------
2016-07-16 00:48:02,724:INFO:connectionpool.py:788:Starting new HTTPS connection (1): api.reddit.com
2016-07-16 00:48:02,734:INFO:objects.py:193:Loader caught: CannotOverwriteExistingCassetteException - No match for the request (<Request (GET) https://api.reddit.com/r/aww/.json?limit=1024>) was found. Can't overwrite existing cassette ('/home/mardi/repos/rtv/tests/cassettes/test_subreddit_frontpage_toggle.yaml') in your current record mode ('none').
===================== 1 failed, 109 passed, 1 xfailed in 24.18 seconds ======================

Here is the test I added:
```
def test_subreddit_frontpage_toggle(subreddit_page, terminal):
    with mock.patch.object(terminal, 'prompt_input'):
        terminal.prompt_input.return_value = 'aww'
        subreddit_page.controller.trigger('/')
        assert subreddit_page.content.name == '/r/aww'
        subreddit_page.controller.trigger('p')
        assert subreddit_page.content.name == '/r/front'
```